cibuildwheel/linux.py

```diff
@@ -129,72 +117,34 @@ def build(project_dir, package_name, output_dir, test_command, test_requires, be
-
-        timeout = 180
-        start = monotonic.monotonic()
-        while True:
-            time.sleep(1)
-            command = [
-                'docker',
-                'cp',
-                '{}/.'.format(os.path.abspath(project_dir)),
-                '{}:/project'.format(container_name),
-            ]
-            print('docker command: {}'.format(command))
-            if subprocess.call(command) == 0:
-                break
-
-            if monotonic.monotonic() - start > timeout:
-                raise DockerRunTimeoutError(
-                    'Unable to successfully copy project directory'
-                    ' within {} seconds'.format(timeout)
-                )
+        def run_docker(command, stdin_str=None):
+            print('docker command: docker {}'.format(' '.join(map(shlex_quote, command))))
+            if stdin_str is None:
+                subprocess.check_call(['docker'] + command)
+            else:
+                process = subprocess.Popen(['docker'] + command,
+                                           stdin=subprocess.PIPE, universal_newlines=True)
+                try:
+                    process.communicate(stdin_str)
+                except KeyboardInterrupt:
+                    process.kill()
+                    process.wait()
+                if process.returncode != 0:
+                    raise subprocess.CalledProcessError(process.returncode, process.args)
 
+        container_name = 'cibuildwheel-{}'.format(uuid.uuid4())
         try:
-            docker_process.communicate(bash_script)
-        except KeyboardInterrupt:
-            docker_process.kill()
-            docker_process.wait()
-
-        command = [
-            'docker',
-            'cp',
-            '{}:/output/.'.format(container_name),
-            os.path.abspath(output_dir),
-        ]
-        print('docker command: {}'.format(command))
-        subprocess.check_call(command)
-
-        command = [
-            'docker',
-            'rm',
-            '-v', container_name,
-        ]
-        print('docker command: {}'.format(command))
-        subprocess.check_call(command)
-
-        if docker_process.returncode != 0:
+            run_docker(['create',
+                        '--env', 'CIBUILDWHEEL',
+                        '--name', container_name,
+                        '-i',
+                        '-v', '/:/host', # ignored on Circle
+                        docker_image, '/bin/bash'])
+            run_docker(['cp', os.path.abspath(project_dir) + '/.', container_name + ':/project'])
+            run_docker(['start', '-i', '-a', container_name], stdin_str=bash_script)
+            run_docker(['cp', container_name + ':/output/.', os.path.abspath(output_dir)])
+        except subprocess.CalledProcessError:
             exit(1)
+        finally:
+            # Still gets executed, even when 'exit(1)' gets called
+            run_docker(['rm', '-v', container_name])
```

Comment:
Add the [`--force`](https://docs.docker.com/engine/reference/commandline/rm/) option to this? I don't know if it's possible for it still to be running at this point but it can't hurt I think

Reply:
Uuuh, right. I think I forgot to transfer that from the old code :-/